src/shared/components/__tests__/RichTable.unit.test.tsx

```diff
@@ -0,0 +1,2092 @@
+          });
+
+          expect(result.current.page).toBe(1);
+        });
+      });
+    });
+  });
+
+  describe("header", () => {
+    const TestComponent = ({ cells }: { cells: RichTableCell[] }) => (
+      <table>
+        <thead>
+          <tr>
+            {cells.map((cell) => (
+              <th data-testid="cell" {...cell.useCellProps()}>
+                {cell.render()}
+              </th>
+            ))}
+          </tr>
+        </thead>
+      </table>
+    );
+
+    it("renders the column header", async () => {
+      const { result, waitForNextUpdate } = renderHook(() =>
+        useRichTable<TestData>({
+          columns: [{ key: "id", header: "ID" }],
+          list: apiWithoutPagination,
+          paginated: false,
+        })
+      );
+      await waitForNextUpdate();
+
+      const { getByText } = render(
+        <TestComponent cells={result.current.header.cells} />
+      );
+
+      expect(getByText("ID")).toBeInTheDocument();
+    });
+
+    it("only renders provided columns", async () => {
+      const { result, waitForNextUpdate } = renderHook(() =>
+        useRichTable<TestData>({
+          columns: [{ key: "id", header: "ID" }],
+          list: apiWithoutPagination,
+          paginated: false,
+        })
+      );
+      await waitForNextUpdate();
+
+      const { queryAllByTestId } = render(
+        <TestComponent cells={result.current.header.cells} />
+      );
+
+      expect(queryAllByTestId("cell").length).toBe(1);
+    });
+
+    it("renders a checkbox when the table is selectable", async () => {
+      const { result, waitForNextUpdate } = renderHook(() =>
+        useRichTable<TestData>({
+          columns: [{ key: "id", header: "ID" }],
+          list: apiWithoutPagination,
+          paginated: false,
+          selectable: true,
+        })
+      );
+      await waitForNextUpdate();
+
+      const { queryByRole } = render(
+        <TestComponent cells={result.current.header.cells} />
+      );
+
+      expect(queryByRole("checkbox")).not.toBeNull();
+    });
+
+    it("sets the width of each column if provided", async () => {
+      const { result, waitForNextUpdate } = renderHook(() =>
+        useRichTable<TestData>({
+          columns: [{ key: "id", header: "ID", width: 25 }],
+          list: apiWithoutPagination,
+          paginated: false,
+        })
+      );
+      await waitForNextUpdate();
+
+      const { getByTestId } = render(
+        <TestComponent cells={result.current.header.cells} />
+      );
+
+      expect(getByTestId("cell")?.style.width).toBe("25%");
+    });
+
+    it("returns a valid object for useRowProps()", async () => {
+      const { result, waitForNextUpdate } = renderHook(() =>
+        useRichTable<TestData>({
+          columns: [{ key: "id", header: "ID" }],
+          list: apiWithoutPagination,
+          paginated: false,
+        })
+      );
+      await waitForNextUpdate();
+
+      render(<div {...result.current.header.useRowProps()} />);
```

Comment:
This test should have an `expect` in it for clarity.

Reply:
4b7a148